apps/src/StudioApp.js

```diff
@@ -1266,11 +1371,15 @@ StudioApp.prototype.builderForm_ = function(onAttemptCallback) {
 */
 StudioApp.prototype.report = function(options) {
   // copy from options: app, level, result, testResult, program, onComplete
-  var report = options;
-  report.pass = this.feedback_.canContinueToNextLevel(options.testResult);
-  report.time = ((new Date().getTime()) - this.initTime);
-  report.attempt = this.attempts;
-  report.lines = this.feedback_.getNumBlocksUsed();
+  var report = $.extend({}, options, {
+    pass: this.feedback_.canContinueToNextLevel(options.testResult),
+    time: ((new Date().getTime()) - this.initTime),
+    attempt: this.attempts,
+    lines: this.feedback_.getNumBlocksUsed(),
+  });
```

Comment:
Somewhat less interesting, but with ES6 we could start using the spread operator instead of $.extend, for example
```
var report = {
  ...options,
  ...{
    pass: this.feedback_.canContinueToNextLevel(options.testResult),
    time: ((new Date().getTime()) - this.initTime),
    attempt: this.attempts,
    lines: this.feedback_.getNumBlocksUsed(),
  })
}